spec/models/course/forum/topic_spec.rb

```diff
@@ -37,6 +37,18 @@
         end
       end
 
+      context 'when reinitializing a topic' do
+        subject do
+          test_topic = Course::Forum::Topic.new(posts_attributes: [title: nil])
+          test_topic.save
```

Comment:
It's not that after_initialize is triggered twice. It's triggered once. But the assignment of attributes creates another post before the after_initialize runs.